File: app/src/main/java/bg/o/sim/colourizmus/view/PickerFragment.java
package bg.o.sim.colourizmus.view;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.NumberPicker;

import bg.o.sim.colourizmus.R;
import bg.o.sim.colourizmus.model.CR;


public class PickerFragment extends Fragment {

    private NumberPicker pickerRed, pickerGreen, pickerBlue;

    public static Fragment newInstance() {
        return new PickerFragment();
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);

        CR.LIVE_COLOUR.observe(this, integer -> {
            pickerRed.setValue(CR.LIVE_COLOUR.getRed());
            pickerGreen.setValue(CR.LIVE_COLOUR.getGreen());
            pickerBlue.setValue(CR.LIVE_COLOUR.getBlue());
        });
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_picker, container, false);

        pickerRed = view.findViewById(R.id.red_picker);
        pickerGreen = view.findViewById(R.id.green_picker);
        pickerBlue = view.findViewById(R.id.blue_picker);

        pickerRed.setMaxValue(255);
        pickerGreen.setMaxValue(255);
        pickerBlue.setMaxValue(255);

        pickerRed.setOnValueChangedListener((picker, oldVal, newVal) -> CR.LIVE_COLOUR.setRed(newVal));
        pickerGreen.setOnValueChangedListener((picker, oldVal, newVal) -> CR.LIVE_COLOUR.setGreen(newVal));
        pickerBlue.setOnValueChangedListener((picker, oldVal, newVal) -> CR.LIVE_COLOUR.setBlue(newVal));

        return view;
    }

}

File: app/src/main/java/bg/o/sim/colourizmus/view/PickerFragment.kt
package bg.o.sim.colourizmus.view

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import bg.o.sim.colourizmus.databinding.FragmentPickerBinding
import bg.o.sim.colourizmus.model.LIVE_COLOUR


class PickerFragment : Fragment() {

    private lateinit var binding: FragmentPickerBinding

    companion object {
        fun newInstance() = PickerFragment()
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)

        LIVE_COLOUR.observe(viewLifecycleOwner) {
            binding.redPicker.value = LIVE_COLOUR.getRed()
            binding.greenPicker.value = LIVE_COLOUR.getGreen()
            binding.bluePicker.value = LIVE_COLOUR.getBlue()
        }
    }


    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        binding = FragmentPickerBinding.inflate(inflater)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        binding.redPicker.maxValue = 255
        binding.greenPicker.maxValue = 255
        binding.bluePicker.maxValue = 255

        binding.redPicker.setOnValueChangedListener { _, _, newVal -> LIVE_COLOUR.setRed(newVal) }
        binding.greenPicker.setOnValueChangedListener { _, _, newVal -> LIVE_COLOUR.setGreen(newVal) }
        binding.bluePicker.setOnValueChangedListener { _, _, newVal -> LIVE_COLOUR.setBlue(newVal) }
    }
}
